helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -560,6 +560,28 @@ private void processMaintenanceMode(String clusterName, final boolean enabled,
     }
   }
 
+  private enum ResetPartitionFailureReason {
+    INSTANCE_NOT_ALIVE("%s is not alive in cluster %s"),
+    INSTANCE_NON_EXISTENT("%s does not exist in cluster %s"),
+    RESOURCE_NON_EXISTENT("resource %s is not added to cluster %s"),
+    PARTITION_NON_EXISTENT("not all %s exist in cluster %s"),
+    PARTITION_NOT_ERROR("%s is NOT found in cluster %s"),
+    STATE_MODEL_NON_EXISTENT("%s is NOT found in cluster %s"),
+    PENDING_MSG("a pending message %s exists for resource %s");
```

Comment:
Let's don't bother including PENDING_MSG here. This is not compatible with the other types. And if there is no other better way, we should not be blocked on this.

Just leave it as it originally be.

Reply:
Updated. THX